File: ORB.Core/Models/UserProject.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public class UserProject : EntityWithEditor
    {
        public int UserID { get; set; }
        public int ProjectID { get; set; }
        public bool IsAdmin { get; set; } 
        public bool IsRiskAdmin { get; set; }

        public virtual Project Project { get; set; }
        public virtual User User { get; set; }
    }
}

File: corporate-reporting/src/types/UserProject.ts
import { Entity, IEntity } from "./Entity";
import { IUser } from "./User";
import { IProject } from "./Project";

export interface IUserProject extends IEntity {
    UserID: number;
    ProjectID: number;
    IsAdmin: boolean;
    IsRiskAdmin: boolean;
    User?: IUser;
    Project?: IProject;
}

export class UserProject extends Entity implements IUserProject {
    public UserID = null;
    public ProjectID = null;
    public IsAdmin = false;
    public IsRiskAdmin = false;
}